### DISCUSSION TITLE
What happens to assister if assistedflash is true, but, also someone did enough damage for an ast?

### DISCUSSION THREAD

**Discussion by dbalders** [saul/demofile#262] [Questions] [answered]
**Is your feature request related to a problem? Please describe.**
Hey. On player death, there is the `assister` field, along with `assistedflash` boolean. But, technically 3 people could be involved in a kill, with the attacker, damage assister, and flash assister. Is there a way to handle these cases other than capturing everything on `player_hurt` and calculating later? Or is that the best way to do it?

**Describe the solution you'd like**
Possibly change the assistedflash field from boolean to the flasher's userid? and then assister can just be damage assists? Idk.

**Comment by saul** [comment]
> Possibly change the assistedflash field from boolean to the flasher's userid

The events are defined by the game - this library does no determination of who assisted in kills. This library just exposes the information that's available in the demo file itself.

As a rule, events will contain just enough information for the client. Very rarely is extra information (that is not used by the client) available. The purpose of the player death events is to populate the kill feed in the top-right corner of each player's screen.

**Comment by saul** [comment, marked as answer]
To answer your original question, I suggest you figure out if `assistedflash` is `true` when the flash assister is the same player mentioned in `assister`. Then you can figure out if you need to keep track of damage assists in `player_hurt`, or flash assists in `player_blind` events.

**Comment by dbalders** [comment]
Ok cool, thanks saul! Im fairly sure if assistedflash is true the assister is the flasher, but, Ill make 100% sure and then post my solution back here in case anyone comes across this wanting a solution also.
